runtime/compiler/optimizer/UnsafeFastPath.cpp

```diff
@@ -279,6 +280,26 @@ bool TR_UnsafeFastPath::tryTransformUnsafeAtomicCallInVarHandleAccessMethod(TR::
       if (isVarHandleOperationMethodOnNonStaticField(callerMethod) &&
          performTransformation(comp(), "%s transforming Unsafe.CAS [" POINTER_PRINTF_FORMAT "] into codegen inlineable\n", optDetailString(), node))
          {
+      #if defined(J9VM_GC_ENABLE_SPARSE_HEAP_ALLOCATION)
+         if (isUnsafeCallerAccessingArrayElement(callerMethod) && TR::Compiler->om.isOffHeapAllocationEnabled() && comp()->target().is64Bit())
+            {
+            TR::Node *object = node->getChild(1);
+            TR::Node *offset = node->getChild(2);
+
+            TR::Node *baseAddr = TR::TransformUtil::generateDataAddrLoadTrees(comp(), object);
+
+            unsafeAddress = TR::Node::create(node, TR::aladd, 2, baseAddr, offset);
+            unsafeAddress->setIsInternalPointer(true);
+
+            node->setChild(1, baseAddr);
```

Comment:
Ditto re the tree form.  Plus, you set child(1) to be the `aladd`, what about child(2) remaining in the tree?